pkgs/by-name/ic/icestudio/package.nix

```
@@ -0,0 +1,96 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildNpmPackage,
+  nwjs,
+  makeWrapper,
+  python3,
+  python3Packages,
+  fetchpatch,
+  fetchurl,
+}:
+
+let
+  # Use unstable because it has improvements for finding python
+  version = "unstable-2024-07-11";
+
+  src = fetchFromGitHub {
+    owner = "FPGAwars";
+    repo = "icestudio";
+    rev = "5df95bfbcc9540675897256e0a372f8039b534c5";
+    hash = "sha256-0HETAeR8OCsu7caP8SbBVYSLQIxMKjV1flR24MizSnU=";
+  };
+
+  collection = fetchurl {
+    url = "https://github.com/FPGAwars/collection-default/archive/v0.4.1.zip";
+    hash = "sha256-F2cAqkTPC7xfGnPQiS8lTrD4y34EkHFUEDPVaYzVVg8=";
+  };
+
+  app = buildNpmPackage {
+    pname = "icestudio-app";
+    inherit version src;
+    sourceRoot = "${src.name}/app";
+    npmDepsHash = "sha256-gfkLW8OaCpaq5KHBhttqMTQPfhUCs22ii6UvNxqWnsQ=";
+    dontNpmBuild = true;
+    installPhase = ''
+      cp -r . $out
+    '';
+  };
+in
+buildNpmPackage rec {
+  inherit version src;
+  pname = "icestudio";
+  npmDepsHash = "sha256-K90zgKHqNEvjjM20YaXa24J0dGA2MESNvzvsr31AR2Y=";
+  npmFlags = [
+    # Use the legacy dependency resolution, with less strict version requirements for transative dependencies
+    "--legacy-peer-deps"
+
+    # We want to avoid call the scripts/postInstall.sh until we copy the collection and app derivation we do that on installPhase.
+    "--ignore-scripts"
```

Wrapping comments makes them more readable:

```suggestion
    # Use the legacy dependency resolution, with less strict version
    # requirements for transative dependencies
    "--legacy-peer-deps"

    # We want to avoid call the scripts/postInstall.sh until we copy the
    # collection and app derivation we do that on installPhase
    "--ignore-scripts"
```